### ISSUE TITLE
Bug in Polycyclic package when applying OrbitIntegralAction to an example group

### ISSUE THREAD

**Issue by rahtol** [gap-system/gap#2179]
### Observed behaviour

Using Gap4r8 and package Polycyclic-2.11 I applied the function `OrbitIntegralAction` to an example Group
that comes with the package. See the gap code below
```gap
# create PcpGroup 

# 8 with 5 generators from examples library that comes with polycyclic package
# (see gap/exam/pcplib.gi)
G:=ExamplesOfSomePcpGroups(8);
# create a 7by7 integral matrix representation of the group
# (see gap/matrep/matrep.gi in polycyclic package)
actG:=Representation(Pcp(G)!.gens[1]!.collector);
# check the relations of the polycyclic representation on the matrix
# representation to make sure the example is consistent
Print("b^a = b*d^-1 : ", actG[2]^actG[1]=actG[2]*actG[4]^-1, "\n");
Print("c^b = e : ", actG[3]^actG[2]=actG[5], "\n");
Print("d^b = c*d^-1*e : ", actG[4]^actG[2]=actG[3]*actG[4]^-1*actG[5], "\n");
Print("e^b = d : ", actG[5]^actG[2]=actG[4], "\n");
# a first randomly chosen example vector
v:=[8,-4,5,2,13,-17,9];
# construct a second vector in the orbit of v
exp:=[ -2, 2, 0, 5, 5 ];
w := v * MappedVector( exp, actG );
# ready to call - expecting o.prei on return to be a valid element
# of G with v * o.prei = w 
o := OrbitIntegralAction( G, actG, v, w);
```

The call to OrbitIntegral Action does not terminate normally. Instead I get the following Error:
```
gap> rd("matrix50.g");
b^a = b*d^-1 : true
c^b = e : true
d^b = c*d^-1*e : true
e^b = d : true
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `[]:=' on 3 arguments called from
v[l
 ] at C:/tools/cygwin64/home/bec/gap4r8/pkg/polycyclic-2.11/gap/matrix/rowbase\
s.gi:85 called from
CoefficientsByNHSEB( vec, hom
 ) at C:/tools/cygwin64/home/bec/gap4r8/pkg/polycyclic-2.11/gap/matrix/rowbase\
s.gi:106 called from
ImageByNHSEB( x, nath
 ) at C:/tools/cygwin64/home/bec/gap4r8/pkg/polycyclic-2.11/gap/action/orbstab\
.gi:374 called from
func( C[i] ) at C:/tools/cygwin64/home/bec/gap4r8/lib/coll.gi:746 called from
List( derS, function ( x )
      return ImageByNHSEB( x, nath );
  end
 ) at C:/tools/cygwin64/home/bec/gap4r8/pkg/polycyclic-2.11/gap/action/orbstab\
.gi:374 called from
...  at line 16 of c:/tools/cygwin64/home/bec/gapwork/matrix50.g
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk>
```
### Expected behaviour

I would have expected OrbitIntegral to terminate normally and deliver some g in G with v*g=w.

### Copy and paste GAP banner (to tell us about your setup)
```
 *********   GAP 4.8.6, 12-Nov-2016, build of 2017-02-13 23:12:16 (CET)
 *  GAP  *   http://www.gap-system.org
 *********   Architecture: x86_64-unknown-cygwin-gcc-default64
 Libs used:  gmp
 Loading the library and packages ...
 Components: trans 1.0, prim 2.1, small* 1.0, id* 1.0
 Packages:   AClib 1.2, Alnuth 3.0.0, AtlasRep 1.5.1, AutPGrp 1.6,
             CRISP 1.4.4, Cryst 4.1.12, CrystCat 1.1.6, CTblLib 1.2.2,
             FactInt 1.5.3, FGA 1.3.1, GAPDoc 1.5.1, IRREDSOL 1.3.1,
             LAGUNA 3.7.0, Polenta 1.3.7, Polycyclic 2.11, RadiRoot 2.7,
             ResClasses 4.5.0, Sophus 1.23, SpinSym 1.5, TomLib 1.2.6,
             Utils 0.43
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap>
```

### my analysis of the problem

Line numbers in the following are with respect to file gap/action/orbstb.gi,
especially in function OrbitCongruenceAction ranging from line 342 to 436

The error ocurs primarily in line 374 because derS[1] is not in the domain (nath.source) of
the NaturalHomomorphismBySemiEchelonBases( ser[i], ser[i+1] ).

**Comment by rahtol** [comment]
[matrix50.zip](https://github.com/gap-system/gap/files/1724111/matrix50.zip)
matrix50.g GAP code constructing the example and calling OrbitIntegralAction
matrix50.txt  Stackdump of OrbitIntegralAction after failure

**Comment by fingolfin** [comment]
Hi, thanks for your report. Some remarks:
* please don't ever write `Pcp(G)!.gens[1]!.collector` -- this may well stop working in future versions of  polycyclic. Instead, just write `Collector(G)`
* I suspect this is related to https://github.com/gap-packages/polycyclic/issues/15
* please submit any further issues for the polycclic package to https://github.com/gap-packages/polycyclic/issues/  and not to the general GAP bug tracker.

Thanks!

**Comment by fingolfin** [comment]
Oh, and also note that `MappedVector` is undocumented, so use it at your own risk.

**Comment by fingolfin** [comment]
Logged this issue now under https://github.com/gap-packages/polycyclic/issues/21

**Comment by fingolfin** [comment]
This will be fixed in the polycyclic release, which hopefully will come this week.

**Comment by fingolfin** [comment]
This has been fixed in Polycyclic 2.12, which you can download from https://gap-packages.github.io/polycyclic/
